src/main/java/hudson/remoting/Launcher.java

```diff
@@ -516,19 +516,34 @@ public static void main(InputStream is, OutputStream os, Mode mode, boolean perf
         if (os instanceof StandardOutputStream)
             cb.withProperty(StandardOutputStream.class,os);
 
-        Channel channel = cb.build(is, os);
+        final Channel channel = cb.build(is, os);
         System.err.println("channel started");
         long timeout = 1000 * Long.parseLong(
                 System.getProperty("hudson.remoting.Launcher.pingTimeoutSec", "240")),
              interval = 1000 * Long.parseLong(
                 System.getProperty("hudson.remoting.Launcher.pingIntervalSec", "600"));
+
+        boolean pingFailureAnalyzer = Boolean.parseBoolean(System.getProperty("hudson.remoting.Launcher.pingFailureAnalyzer", "false"));
+
         if (performPing && timeout > 0 && interval > 0) {
-            new PingThread(channel, timeout, interval) {
+            new PingThread(channel, timeout, interval, pingFailureAnalyzer) {
```

Comment:
there are more ways to create/construct the PingThread than just here - why not put this in the base class - or in a class that the PingThread can call?